spec/requests/partner_users_requests_spec.rb

```diff
@@ -2,20 +2,40 @@
 
 RSpec.describe PartnerUsersController, type: :request do
   let!(:partner) { create(:partner) } # Assuming you have a factory for creating partners
-  let!(:user) { create(:user) } # Assuming you have a factory for creating users
+  let(:organization) { create(:organization) }
+  let(:user) { create(:user, organization: organization) }
+  let(:org_admin) { create(:organization_admin, organization: organization) }
   let(:default_params) do
     {organization_id: @organization.to_param}
   end
 
-  before do
-    sign_in(user)
+  shared_examples "denies access" do
```

Comment:
The example is only 2 lines long. I generally find that shared examples make things a lot harder to debug since the backtrace gets a lot more confusing. I try to use them only if I really get some significant benefit out of them. Can we leave this WET? :smile: